### PR TITLE
Add a manual run_tl trigger

### PR DESCRIPTION
**Pull request by sastels**
add a "run_tl" command to `post_to_endpoint.py`.

### CODE REVIEW

File: extract_endpoint/tests/test_post_to_endpoint.py [hunk 1]
```diff
@@ -150,3 +151,14 @@ def test_post_stream_cli_no_url(azure_service: blob.BlockBlobService,
...
+
+
+@pytest.mark.usefixtures('run_endpoint')
+def test_run_tl() -> None:
+    runner = testing.CliRunner()
+    result = runner.invoke(post_to_endpoint.main, args=[
+        'run_tl',
+        '--url', None
+    ])
+    assert result.exit_code == HTTPStatus.BAD_GATEWAY
```
Comment (sastels) [inline extract_endpoint/tests/test_post_to_endpoint.py]: getting a `BAD_GATEWAY` here means that Post connected successfully to Endpoint and supplied the correct salt and signature, but Endpoint could not connect to the TL web app. Essentially, Post did what it was supposed to.

File: extract_endpoint/src/extract_endpoint/endpoint.py [hunk 2]
```diff
@@ -29,8 +29,8 @@
 ))
 
 
-def _trigger_url() -> str:
-    return os.environ.get('TRIGGER_ADDRESS', 'http://0.0.0.0:5010') + '/run_tl'
+def _run_tl_url() -> str:
+    return os.environ.get('TL_ADDRESS', 'http://0.0.0.0:5010') + '/run_tl'
```
Comment (buckley-w-david) [inline extract_endpoint/src/extract_endpoint/endpoint.py]: Has the environment variable name been updated in Azure/Is @szinck1 aware of the change?

Comment (sastels) [reply extract_endpoint/src/extract_endpoint/endpoint.py]: He hasn't added either to Azure yet. (still need to do that and proper Mongo keys)

Comment (buckley-w-david) [reply extract_endpoint/src/extract_endpoint/endpoint.py]: He is adding the database variables as we speak

File: extract_endpoint/src/extract_endpoint/post_to_endpoint.py [hunk 3]
```diff
@@ -31,10 +36,32 @@ def post_stream(stream: typing.IO[bytes],
...
+    return requests.post(url, data=data).status_code
+
+
+def trigger_tl(url) -> int:
+    salt = secrets.token_hex(16)
+    hasher = hashlib.new('sha3_256')
+    hasher.update((salt + _etl_secret_key()).encode())
+    signature = hasher.hexdigest()
+    data = dict(salt=salt, signature=signature)
+    return send_to_tl_app(url, data)
```
Comment (buckley-w-david) [inline extract_endpoint/src/extract_endpoint/post_to_endpoint.py]: I don't really see the point in wrapping just a single POST in a function call, any reasoning behind that?

File: extract_endpoint/src/extract_endpoint/post_to_endpoint.py [hunk 4]
```diff
@@ -31,10 +36,32 @@ def post_stream(stream: typing.IO[bytes],
                          data={'salt': salt, 'signature': signature, 'timestamp': timestamp})
 
 
+def send_to_tl_app(url: str, data: typing.Dict[str, str]) -> int:
+    return requests.post(url, data=data).status_code
```
Comment (buckley-w-david) [inline extract_endpoint/src/extract_endpoint/post_to_endpoint.py]: You wrap your other posts in `try/except` blocks, but not this one. Intentional?

File: extract_endpoint/src/extract_endpoint/post_to_endpoint.py (hunk 3 above)
Comment (buckley-w-david) [reply extract_endpoint/src/extract_endpoint/post_to_endpoint.py]: Also, it seems odd to have this much functionality in the the CLI file. Should this stuff be refactored into it's own file, and the CLI left to just do CLI stuff?

File: extract_endpoint/src/extract_endpoint/post_to_endpoint.py (hunk 4 above)
Comment (sastels) [reply extract_endpoint/src/extract_endpoint/post_to_endpoint.py]: no - should do this one too. good catch.

File: extract_endpoint/src/extract_endpoint/post_to_endpoint.py (hunk 3 above)
Comment (sastels) [reply extract_endpoint/src/extract_endpoint/post_to_endpoint.py]: I figured they were below the threshold for doing that, ie they're just doing a hash and a post.

File: extract_endpoint/src/extract_endpoint/post_to_endpoint.py [hunk 5]
```diff
@@ -46,22 +41,32 @@ def trigger_tl(url) -> int:
...
-    return send_to_tl_app(url, data)
+    try:
+        return requests.post(url, data=data).status_code
+    except requests.exceptions.ConnectionError:
+        return HTTPStatus.BAD_GATEWAY
+
+
+@click.group()
+def main() -> None:
+    pass
```
Comment (buckley-w-david) [inline extract_endpoint/src/extract_endpoint/post_to_endpoint.py]: I like that now all the CLI stuff is at least together

**Pull request merged**